### ISSUE TITLE
[Feature] Pre-release/ Beta release for pull requests

### ISSUE THREAD

**Issue by Tush-r** [RipMeApp/ripme#1766]
## Issue 
Currently the ripper is seeing fewer releases, which is understandable. A lot of community contributions which is nice.
The issue is that not everyone can `git clone` the project and build the changes that fixes their issue. So someone has to come along, build the project and share it( like [this](https://github.com/RipMeApp/ripme/issues/1719#issuecomment-680696082)).

#### *Adding releases for pull requests as beta releases.*
The pull request creator will see a new section in the pr-template, like,

```
[x] Add a release for my changes as a `Beta Release`
```
The users can find the beta releases:
* At the bottom of the pull request page( like the `checks api` box)
* In the release section( of which I'm not 100% sure is possible).
* Within the `ripme` app itself( which again I'm not sure about).

## Working
* Create a new project `ripme-pr-release` which will be deployed on heroku.
* Add a webhook for the heroku app to `ripme`
* The webhook is called on new pull requests, builds the app with a beta flag and add it to one of the places mentioned above.
* The beta release version would contain the current version and the pull-request number( like `1.8.88.1234`).
* Also adding a `download at your own risk...` message would be good.
---

I have this figured out in my head, just need to get it down in code and test.
For all sorts of suggestion or questions, feel free to drop a message.

**Comment by cyian-1756** [comment]
Seems like a good idea to me.

**Comment by soloturn** [comment]
github actions would allow this as well nowadays: https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows.

one could then publish the binary as asset. then take the asset and publish it as release. i did this here - but its not a good example as the source is not in the same repo: https://github.com/soloturn/swift-aur/actions.

**Comment by soloturn** [comment]
i gave it at try in #1776 - mind checking @Tush-r ?

**Comment by cyian-1756** [comment]
> Within the ripme app itself( which again I'm not sure about).

Shouldn't be too hard to add a update to beta option in the config menu. Probably have to disable the hash check in the updater for betas (unless there's a way to update ripme.json automatically).

**Comment by rautamiekka** [comment]
> > Within the ripme app itself( which again I'm not sure about).
> 
> Shouldn't be too hard to add a update to beta option in the config menu. Probably have to disable the hash check in the updater for betas (unless there's a way to update ripme.json automatically).

I don't see why it wouldn't be possible to do automatically.

Granted, I dunno how you do it currently, but SURELY it's possible to make at least 98% automatic.

**Comment by Tush-r** [comment]
> i gave it at try in #1776 - mind checking @Tush-r ?
> 
>     * original build, no PR, no jar: https://github.com/soloturn/ripme/actions/runs/330582439
> 
>     * PR to master, zipped jars uploaded: https://github.com/RipMeApp/ripme/actions/runs/330582473

This looks good, but ripme release requires a few more steps before building the jar. Check `patch.py` and `release.py`.
I'm not familiar with github workflow and actions. I'll take a look at this issue in a few days.

**Comment by soloturn** [comment]
> > i gave it at try in #1776 - mind checking @Tush-r ?
> > ```
> > * original build, no PR, no jar: https://github.com/soloturn/ripme/actions/runs/330582439
> > 
> > * PR to master, zipped jars uploaded: https://github.com/RipMeApp/ripme/actions/runs/330582473
> > ```
> 
> This looks good, but ripme release requires a few more steps before building the jar. Check `patch.py` and `release.py`.
> I'm not familiar with github workflow and actions. I'll take a look at this issue in a few days.

good point, to update the verison. @Tush-r, can you please send me or here a pull request for patch.py to update the version to what you think it should be? then we already would have 2 bricks in place. as patch.py has no real parameter, one could feed in the commit message but thats not good enough:
```
git fetch origin pull/1776/head:pr-1776
git checkout pr-1776
commitmessage=`git log -1 HEAD --pretty=format:%s`
echo $commitmessage | python patch.py
```

**Comment by Tush-r** [comment]
> > > i gave it at try in #1776 - mind checking @Tush-r ?
> > > ```
> > > * original build, no PR, no jar: https://github.com/soloturn/ripme/actions/runs/330582439
> > > 
> > > * PR to master, zipped jars uploaded: https://github.com/RipMeApp/ripme/actions/runs/330582473
> > > ```
> > 
> > 
> > This looks good, but ripme release requires a few more steps before building the jar. Check `patch.py` and `release.py`.
> > I'm not familiar with github workflow and actions. I'll take a look at this issue in a few days.
> 
> good point, to update the verison. @Tush-r, can you please send me or here a pull request for patch.py to update the version to what you think it should be? then we already would have 2 bricks in place. as patch.py has no real parameter, one could feed in the commit message but thats not good enough:
> 
> ```
> git fetch origin pull/1776/head:pr-1776
> git checkout pr-1776
> commitmessage=`git log -1 HEAD --pretty=format:%s`
> echo $commitmessage | python patch.py
> ```

Not sure how to proceed with it. Creating custom actions maybe one way to go about it. I'll create a PR on your fork, if I get something working in a few days.
